sched/signal/sig_action.c

```diff
@@ -102,7 +102,7 @@ static FAR sigactq_t *nxsig_alloc_action(void)
 
   /* Check if we got one. */
 
-  while (!sigact)
+  if (!sigact)
```

Comment:
keep while loop, there are chance other thread grab the free action after line 109 and before line 113

Reply:
thanks, will do.